### ISSUE TITLE
dockerTools the architecture of the generated image is amd64 while building on aarch64

### ISSUE THREAD

**Issue by hughjfchen** [NixOS/nixpkgs#87345]
**Arch of the generated image is amd64 while building on aarch64**
The architecture of the generated image is `amd64` while the image is built on an aarch64 platform.
That makes users cannot download the image for aarch64 platform.

**To Reproduce**
1. build a image with `buildImage` on an `aarch64` platform, like a RPi4
2. load the image up and inspect the image
3. the `Architecture` of the image is `amd64`

**Expected behavior**
The `Architecture` of the generated image should be `arm64`

**Screenshots**

**Additional context**

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
> chenjf@orangepi4:~$ nix-shell -p nix-info --run "nix-info -m"
> these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
>   /nix/store/0ryg3zjgyq0gy24qnbk86is595r67xwr-nix-info
> copying path '/nix/store/0ryg3zjgyq0gy24qnbk86is595r67xwr-nix-info' from > 'https://cache.nixos.org'...
> bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
> /nix/store/nvv874lqz94nbyfl0fxqmcwgjpgs2xvr-bash-4.4-p23/bin/bash: warning: setlocale: LC_ALL: > cannot change locale (en_US.UTF-8)
>  - system: `"aarch64-linux"`
>  - host os: `Linux 5.4.32-rockchip64, Ubuntu, 18.04.4 LTS (Bionic Beaver)`
>  - multi-user?: `no`
>  - sandbox: `yes`
>  - version: `nix-env (Nix) 2.3.4`
>  - channels(chenjf): `"nixpkgs-19.09.2477.a7ceb2536ab"`
>  - nixpkgs: `/home/chenjf/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by dashmoho** [comment]
I believe this is already fixed in 20.03 by 8166bc934b254d61ea059f63764632fc2785d9da. Can you try this version?

**Comment by hughjfchen** [comment]
Yes, it's been fixed in 20.03. Thanks for your help. I'll close this issue.
